iree/compiler/Dialect/Flow/Transforms/test/form_streams.mlir

```diff
@@ -1,5 +1,63 @@
 // RUN: iree-opt -split-input-file -iree-flow-form-streams %s | IreeFileCheck %s
 
+// CHECK-LABEL: func @outsideTieShape
+func @outsideTieShape(%arg0: tensor<?xi32> {iree.reflection = {}}, %arg1: !shapex.ranked_shape<[?]> {iree.reflection = {}}) -> (tensor<?xi32> {iree.reflection = {}}, !shapex.ranked_shape<[?]> {iree.reflection = {}}) attributes {iree.module.export} {
+  // CHECK: %[[WORKLOAD1:.+]] = constant 1 : i32
+  %c1_i32 = constant 1 : i32
+  // CHECK-NEXT: %[[WORKLOAD0:.+]] = constant 0 : i32
+  %c0_i32 = constant 0 : i32
+  // CHECK-NEXT: %[[WORKLOAD2:.+]] = constant 0 : index
+  %c0 = constant 0 : index
+  // CHECK-NEXT: %[[WORKLOAD3:.+]] = constant 1 : index
+  %c1 = constant 1 : index
+  // CHECK-NEXT: %0 = shapex.ranked_dim %arg1[0] : !shapex.ranked_shape<[?]> -> index
+  %0 = shapex.ranked_dim %arg1[0] : !shapex.ranked_shape<[?]> -> index
+  // CHECK-NEXT: %1 = shapex.make_ranked_shape %[[WORKLOAD2]] : (index) -> !shapex.ranked_shape<[?]>
+  %1 = shapex.make_ranked_shape %c0 : (index) -> !shapex.ranked_shape<[?]>
+  // CHECK-NEXT: shapex.tie_shape %arg0, %arg1 : tensor<?xi32>, !shapex.ranked_shape<[?]>
+  %2 = shapex.tie_shape %arg0, %arg1 : tensor<?xi32>, !shapex.ranked_shape<[?]>
+  // CHECK-NEXT: dim %2, %[[WORKLOAD2]] : tensor<?xi32>
+  %3 = dim %2, %c0 : tensor<?xi32>
+  // CHECK-NEXT: index_cast %3 : index to i32
+  %4 = index_cast %3 : index to i32
+  // CHECK-NEXT: cmpi "slt", %4, %[[WORKLOAD0]] : i32
+  %5 = cmpi "slt", %4, %c0_i32 : i32
+  // CHECK-NEXT: select %5, %4, %[[WORKLOAD0]] : i32
+  %6 = select %5, %4, %c0_i32 : i32
+  // CHECK-NEXT: cmpi "sge", %6, %[[WORKLOAD0]] : i32
+  %7 = cmpi "sge", %6, %c0_i32 : i32
+  // CHECK-NEXT: select %7, %6, %[[WORKLOAD0]] : i32
+  %8 = select %7, %6, %c0_i32 : i32
+  // CHECK-NEXT: addi %6, %[[WORKLOAD1]] : i32
+  %9 = addi %6, %c1_i32 : i32
+  // CHECK-NEXT: cmpi "sge", %8, %9 : i32
+  %10 = cmpi "sge", %8, %9 : i32
+  // CHECK-NEXT: select %10, %8, %9 : i32
+  %11 = select %10, %8, %9 : i32
+  // CHECK-NEXT: cmpi "slt", %11, %4 : i32
+  %12 = cmpi "slt", %11, %4 : i32
+  // CHECK-NEXT: select %12, %11, %4 : i32
+  %13 = select %12, %11, %4 : i32
+  // CHECK-NEXT: %14 = shapex.ranked_dim %arg1[0] : !shapex.ranked_shape<[?]> -> index
+  // CHECK-NEXT: %15 = shapex.ranked_dim %1[0] : !shapex.ranked_shape<[?]> -> index
+  // CHECK-NEXT: %16 = flow.ex.stream.fragment(%arg2 = %arg0 : tensor<?xi32>, %arg3 = %14 : index, %arg4 = %[[WORKLOAD3]] : index, %arg5 = %8 : i32, %arg6 = %13 : i32, %arg7 = %[[WORKLOAD1]] : i32, %arg8 = %[[WORKLOAD2]] : index, %arg9 = %2 : tensor<?xi32>, %arg10 = %0 : index, %arg11 = %15 : index) -> tensor<?xi32> {
+  // CHECK-NEXT:   %17 = shapex.make_ranked_shape %arg3 : (index) -> !shapex.ranked_shape<[?]>
+  // CHECK-NEXT:   %18 = shapex.make_ranked_shape %arg11 : (index) -> !shapex.ranked_shape<[?]>
+  // CHECK-NEXT:   %19 = shapex.tie_shape %arg2, %17 : tensor<?xi32>, !shapex.ranked_shape<[?]>
```

Comment:
The presence of this op is really what the test case is trying to verify, right? Could we please reduce the test case to be much less fragile in terms of CHECKs, relying on general tests for the other invariants? For example, other parts of the infra/tests verify that capture and mapping logic is correct, and the verifier will fail if it is mechanically done wrong. I think I would be fine with just a check that a stream gets produced with a tie_shape that feeds into the proper flow.dispatch (by name). Removing all of the other over-constrained checks will make this more maintainable and robust to incidental changes.